### PR TITLE
[ARM CPU] Fix AUGRU layer in dien.xml model

### PR DESCRIPTION
**Pull request by allnes**
### Details:
 - *Fixed problem with precision in RNN and TensorIterator layers*
 - *Corrected shape calculation of current layers in ACL Scheduler*

### Tickets:
 - CVS-123900
 - CVS-134520

### CODE REVIEW

File: src/plugins/intel_cpu/src/nodes/rnn.cpp [hunk 1]
```diff
@@ -514,6 +514,9 @@ void RNN::configurePortDataTypes() {
         // onednn doesn't have fp16 instance
         inDataTypes[xIdx] = outDataTypes[yIdx] = outDataTypes[hoIdx] = inDataTypes[hIdx] = memory::data_type::f32; // required by oneDNN.
 
+    if (cell_type == dnnl::algorithm::vanilla_augru && inDataTypes[aIdx] == memory::data_type::f16)
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/rnn.cpp:517]: 1. Why we need this change? Is it oneDNN or ACL limitation? Lets add comment with explanation
2. Shouldn't this change be appllied for ARM arch only?
3. Do we have corresponding RNN test instances on ARM?

File: src/plugins/intel_cpu/src/nodes/tensoriterator.cpp [hunk 2]
```diff
@@ -293,7 +293,8 @@ MemoryPtr DynamicBuffer::create_buffer(const dnnl::engine& eng) {
     const auto estimated_iters = estimate_iters();
     const Shape _shape = Shape({count, static_cast<size_t>(abs_stride * estimated_iters), len/elem_size});
     auto _descCreator = BlockedDescCreator::getCommonCreators().at(LayoutType::ncsp);
-    auto new_buffer_desc = _descCreator->createSharedDesc(from->getDesc().getPrecision(), _shape);
+    auto prec = from->getDesc().getPrecision() == ov::element::f16 ? ov::element::f32 : from->getDesc().getPrecision();
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/tensoriterator.cpp:296]: Why we need this change?
TensorIterator has nothing x64 or ARM specific from the impl perspective.

File: src/plugins/intel_cpu/src/nodes/executors/acl/acl_ie_scheduler.cpp [hunk 3]
```diff
@@ -24,8 +24,7 @@ void ACLScheduler::set_num_threads(unsigned int num_threads) {}
 
 void ACLScheduler::schedule_custom(ICPPKernel *kernel, const Hints &hints, const Window &window, ITensorPack &tensors) {
     const Window & max_window = window;
-    const unsigned int num_iterations =
-            max_window.num_iterations(hints.split_dimension()) == 1 ? 1 : max_window.num_iterations_total();
+    const unsigned int num_iterations = max_window.num_iterations(hints.split_dimension());
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/src/nodes/executors/acl/acl_ie_scheduler.cpp:27]: How this issue was found? Shouldn't we extend test coverage?

**Review by dmitry-gorokhov — CHANGES_REQUESTED** [review]

File: src/plugins/intel_cpu/src/nodes/executors/acl/acl_ie_scheduler.cpp (hunk 3 above)
Comment (allnes) [reply src/plugins/intel_cpu/src/nodes/executors/acl/acl_ie_scheduler.cpp:27]: this calculation of iteration is default for all schedulers in ACL, but in documents old way same work

File: src/plugins/intel_cpu/src/nodes/rnn.cpp (hunk 1 above)
Comment (allnes) [reply src/plugins/intel_cpu/src/nodes/rnn.cpp:518]: 1) OneDNN unsupported fp16 precision
2) For all devices where fp16 is used we will have exceptions
3) We don't have instances, but we need to create SLT separation first of all

File: src/plugins/intel_cpu/src/nodes/tensoriterator.cpp (hunk 2 above)
Comment (allnes) [reply src/plugins/intel_cpu/src/nodes/tensoriterator.cpp:296]: this change depend on augru rnn layer if we don't set fp32, seg fault is created in loop after rnn layer

Comment (dmitry-gorokhov) [reply src/plugins/intel_cpu/src/nodes/tensoriterator.cpp:296]: Is doesn't mean we need to create such kind of WA.
TI should work correctly w/o dependency on parent operation. We need to analyze the issue and find real root-cause

Comment (allnes) [reply src/plugins/intel_cpu/src/nodes/tensoriterator.cpp]: its became irrelevant change, dien.xml is worked without tensoriterator changes (all problem was depend on fp16 in rnn and calculate shapes in scheduler)

**Review by alvoron — APPROVED** [review]

**Comment by alvoron** [comment]
Although dien model still can't be inferred successfully on ARM:
```
Check 'element::Type::merge(result_et, get_input_element_type(1), get_input_element_type(2))' failed at src/core/src/op/select.cpp:70:
While validating node 'opset1::Select Select_3046 (opset1::Parameter Parameter_3043[0]:boolean[1,1,10], opset1::Parameter Parameter_3044[0]:f16[1,1,10], opset1::Parameter Parameter_3045[0]:f32[1,1,10]) -> (dynamic[...])' with friendly_name 'Select_3046':
Argument 1 and 2 element types must match.
```
@allnes can we fix this issue in this PR?
Thank you.

**Comment by dmitry-gorokhov** [comment]
> Although dien model still can't be inferred successfully on ARM:
> 
> ```
> Check 'element::Type::merge(result_et, get_input_element_type(1), get_input_element_type(2))' failed at src/core/src/op/select.cpp:70:
> While validating node 'opset1::Select Select_3046 (opset1::Parameter Parameter_3043[0]:boolean[1,1,10], opset1::Parameter Parameter_3044[0]:f16[1,1,10], opset1::Parameter Parameter_3045[0]:f32[1,1,10]) -> (dynamic[...])' with friendly_name 'Select_3046':
> Argument 1 and 2 element types must match.
> ```
> 
> @allnes can we fix this issue in this PR? Thank you.

@allnes  @alvoron  So does Dien works successfuly now or we still have an issue?

**Comment by allnes** [comment]
> > Although dien model still can't be inferred successfully on ARM:
> > ```
> > Check 'element::Type::merge(result_et, get_input_element_type(1), get_input_element_type(2))' failed at src/core/src/op/select.cpp:70:
> > While validating node 'opset1::Select Select_3046 (opset1::Parameter Parameter_3043[0]:boolean[1,1,10], opset1::Parameter Parameter_3044[0]:f16[1,1,10], opset1::Parameter Parameter_3045[0]:f32[1,1,10]) -> (dynamic[...])' with friendly_name 'Select_3046':
> > Argument 1 and 2 element types must match.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @allnes can we fix this issue in this PR? Thank you.
> 
> @allnes @alvoron So does Dien works successfuly now or we still have an issue?

@dmitry-gorokhov dien.xml is worked. 
Problem was created in the PR - https://github.com/openvinotoolkit/openvino/pull/22674
Problem is solved in the PR - https://github.com/openvinotoolkit/openvino/pull/23143
Problem don't depend on current fix

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**